Sources/SwiftCompilerPlugin/PluginMessages.swift

```diff
@@ -0,0 +1,134 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// NOTE: This file should be synced with swift repository.
+// NOTE: Types in this file should be self-contained and should not depend on any non-stdlib types.
+
+internal enum HostToPluginMessage: Codable {
+  case getCapability
+
+  case expandFreestandingMacro(
+    macro: PluginMessage.MacroReference,
+    discriminator: String,
+    syntax: PluginMessage.Syntax
+  )
+
+  case expandAttachedMacro(
+    macro: PluginMessage.MacroReference,
+    macroRole: PluginMessage.MacroRole,
+    discriminator: String,
+    customAttributeSyntax: PluginMessage.Syntax,
+    declSyntax: PluginMessage.Syntax,
+    parentDeclSyntax: PluginMessage.Syntax?
+  )
+}
+
+internal enum PluginToHostMessage: Codable {
+  case expandFreestandingMacroResult(
+    expandedSource: String?,
+    diagnostics: [PluginMessage.Diagnostic]
+  )
+
+  case expandAttachedMacroResult(
+    expandedSources: [String]?,
+    diagnostics: [PluginMessage.Diagnostic]
+  )
+
+  case getCapabilityResult(capability: PluginMessage.PluginCapability)
+}
+
+/*namespace*/ internal enum PluginMessage {
+  static var PROTOCOL_VERSION_NUMBER: Int { 2 }  // Added 'MacroRole.conformance'
+
+  struct PluginCapability: Codable {
+    var protocolVersion: Int
+  }
+
+  static var capability: PluginCapability {
+    PluginCapability(protocolVersion: PluginMessage.PROTOCOL_VERSION_NUMBER)
+  }
+
+  struct MacroReference: Codable {
+    var moduleName: String
+    var typeName: String
+
+    // The name of 'macro' declaration the client is using.
+    var name: String
+  }
+
+  enum MacroRole: String, Codable {
+    case expression
+    case freeStandingDeclaration
+    case accessor
+    case memberAttribute
+    case member
+    case peer
+    case conformance
+  }
+
+  struct SourceLocation: Codable {
+    var fileID: String
+    var fileName: String
+    var offset: Int
```

Comment:
I assume this `offset` and `column` are UTF-8 byte offsets. Could you add doc comments for this? I’ve wondered way too many times before SwiftSyntax added comments for these all over the place.

Same for all the other `offset` fields in this file.